src/cascadia/TerminalConnection/AzureConnection.cpp

```diff
@@ -0,0 +1,478 @@
+
+        //Initialize the request
+        http_request commonRequest(L"POST");
+        commonRequest.set_request_uri(L"common/oauth2/devicecode");
+        utility::string_t body = L"client_id=" + AzureClientID + L"&resource=" + _wantedResource;
+        commonRequest.set_body(body, L"application/x-www-form-urlencoded");
+
+        //Send the request and receive the response as a json value
+        return _requestHelper(loginClient, commonRequest);
+    }
+
+    // Helper function to wait for the user to authenticate using their web browser
+    json::value AzureConnection::_waitForUser(const utility::string_t deviceCode, int pollInterval, int expiresIn)
+    {
+        //Initialize the client
+        http_client pollingClient(_loginUri);
+
+        //Continuously send a poll request until the user authenticates
+        utility::string_t body = L"grant_type=device_code&resource=" + _wantedResource + L"&client_id=" + AzureClientID + L"&code=" + deviceCode;
+        utility::string_t requestUri = L"common/oauth2/token";
+        json::value responseJson;
+        for (int count = 0; count < expiresIn / pollInterval; count++)
+        {
+            // User might close the tab while we wait for them to authenticate, this case handles that
+            if (_closing.load())
+            {
+                throw "Tab closed.";
+            }
+            http_request pollRequest(L"POST");
+            pollRequest.set_request_uri(requestUri);
+            pollRequest.set_body(body, L"application/x-www-form-urlencoded");
+
+            responseJson = _requestHelper(pollingClient, pollRequest);
+
+            if (responseJson.has_field(L"error"))
+            {
+                Sleep(pollInterval * 1000); //Sleep takes arguments in milliseconds
+                continue; //Still waiting for authentication
+            }
+            else
+            {
+                _outputHandlers(winrt::to_hstring("Authenticated.\r\n"));
+                break; //Authentication is done, break from loop
+            }
+        }
+        if (responseJson.has_field(L"error"))
+        {
+            throw "Time out.";
+        }
+        return responseJson;
+    }
+
+    // Helper function to acquire the user's Azure tenants
+    json::value AzureConnection::_getTenants()
+    {
+        //Initialize the client
+        http_client tenantClient(_resourceUri);
+
+        //Initialize the request
+        http_request tenantRequest(L"GET");
+        tenantRequest.set_request_uri(L"tenants?api-version=2018-01-01");
+        tenantRequest.headers().add(L"Accept", L"application/json");
+        tenantRequest.headers().add(L"Authorization", L"Bearer " + _accessToken);
+        tenantRequest.headers().add(L"User-Agent", L"Terminal");
+
+        //Send the request and return the response as a json value
+        return _requestHelper(tenantClient, tenantRequest);
+    }
+
+    json::value AzureConnection::_refreshTokens()
+    {
+        //Initialize the client

```

Comment:
You have an inconsistent comment style. Sometimes you do 
`//foo`
and other times
`// foo`.

I don't know if our formatter checks/fixes that, but our policy is usually the with-a-space version.